ugvc/mrd/srsnv_training_utils.py

```diff
@@ -185,28 +189,23 @@ def prepare_featuremap_for_model(
     if random_seed is not None:
         np.random.seed(random_seed)
     if balanced_sampling_info_fields:
-        # TODO test this code and add a test, I wrote an implementation but didn't even get to run it
-        logger.warning(
-            "Data motif balancing not tested, use at your own risk"
-            f" (balanced_sampling_info_fields={balanced_sampling_info_fields})"
-        )
         # count the number of entries with each combination of info fields to balance by
         balanced_sampling_info_fields_counter = defaultdict(int)
         with pysam.VariantFile(intersect_featuremap_vcf) as fmap:
             for record in fmap.fetch():
                 balanced_sampling_info_fields_counter[
-                    (record.info.get(info_field) for info_field in balanced_sampling_info_fields_counter)
+                    tuple(record.info.get(info_field) for info_field in balanced_sampling_info_fields)
                 ] += 1
         # determine sampling rate per group (defined by a combination of info fields)
         total_size_per_group = total_size // len(balanced_sampling_info_fields_counter)
         downsampling_rate = {
-            k: total_size_per_group / v * overhead_factor for k, v in balanced_sampling_info_fields_counter.items()
+            k: (total_size_per_group / v) * overhead_factor for k, v in balanced_sampling_info_fields_counter.items()
         }
         for k, v in downsampling_rate.items():
             logger.debug(f"downsampling_rate for {k} = {v}")
-            if v > 1:
+            if v > overhead_factor:
                 logger.warning(f"downsampling_rate for {k} = {v} > 1, result will contain less data than expected")
-                downsampling_rate[k] = 1
+                downsampling_rate[k] = min(1, downsampling_rate[k])
```

Comment:
Still not sure what's going on here. sorry...
If we want to make sure ds rate is not larger than 1, why don't we do this explicitly?
downsampling_rate[k] = min(1, downsampling_rate[k])

Reply:
It is done explicitly in the next line